src/market.py

```diff
@@ -183,14 +205,66 @@ def parse_fill_event(self, event) -> FilledOrder:
             fee_cost=event.native_fee_or_rebate * (1 if event.event_flags.maker else -1),
         )
 
-    def place_order(self, order_params: NewOrderParams):
-        pass
+    def place_order(self, order_params: PlaceOrderParams):
```

Comment:
I don't think we need to decode `PlaceOrderParams` so this function can just be,

`place_order(self, payer: PublicKey, owner: Account, ...)`